homeassistant/components/livisi/shc.py

```diff
@@ -0,0 +1,246 @@
+"""Code for the Livisi Smart Home Controller."""
+import asyncio
+from collections.abc import Callable
+import json
+from typing import Any
+
+from aiolivisi import AioLivisi, Websocket
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+
+from .const import (
+    AVATAR_PORT,
+    CLASSIC_PORT,
+    DEVICE_POLLING_DELAY,
+    PSS_DEVICE_TYPE,
+    SWITCH_PLATFORM,
+)
+from .device import Device
+
+
+class SHC:
+    """Represents the Livisi Smart Home Controller."""
+
+    instance = None
+
+    def __init__(self, hass=None, config_entry=None) -> None:
+        """Initialize the Livisi Smart Home Controller."""
+        self._hass: HomeAssistant = hass
+        self._config_entry: ConfigEntry = config_entry
+        self.aiolivisi: AioLivisi = None
+        self._serial_number: str = ""
+        self._controller_type: str = ""
+        self._os_version: str = ""
+        self._devices: list[dict[str, Any]] = []
+        self.included_devices: list[Device] = []
+        self._switch_devices: list[Device] = []
+        self._websocket: Websocket = Websocket()
+        self._is_avatar: bool = False
+        self._port: str = ""
+        self._rooms: dict[str, Any] = {}
+        self._subscribers: dict[str, Callable] = {}
+        if SHC.instance is not None:
+            raise Exception("This class is a singleton!")
```

Comment:
Why is this a singleton if there's only one config entry per SHC instance? If we only allow one config entry per integration we should have a guard in the config flow. Regardless the singleton pattern here looks overkill.